File: app/src/main/java/com/example/aleksei/carmotion/MainActivity.java
package com.example.aleksei.carmotion;

import android.os.Bundle;

import com.example.aleksei.carmotion.car.CarFragment;
import com.example.aleksei.carmotion.car.CarPresenter;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        CarFragment carFragment =
                (CarFragment) getSupportFragmentManager().findFragmentById(R.id.contentFrame);
        if (carFragment == null) {
            carFragment = CarFragment.newInstance();
            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.contentFrame, carFragment)
                    .commit();
        }
        new CarPresenter(carFragment);
    }
}

File: app/src/main/java/com/example/aleksei/carmotion/MainActivity.kt
package com.example.aleksei.carmotion

import android.os.Bundle

import com.example.aleksei.carmotion.view.CarFragment
import com.example.aleksei.carmotion.presenter.CarPresenterImpl

import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var carFragment = supportFragmentManager.findFragmentById(R.id.contentFrame) as CarFragment?
        if (carFragment == null) {
            carFragment = CarFragment.newInstance()
            supportFragmentManager
                .beginTransaction()
                .add(R.id.contentFrame, carFragment)
                .commit()
        }
        CarPresenterImpl(carFragment)
    }
}
